tmpfile and tmpdir creation inside /tmp

iulianbarbu opened the pull request:
Signed-off-by: Iulian Barbu <iul@amazon.com>

Implemented new methods for TempFile struct, respectively for TempDir struct:
- TempFile/TempDir::new() creates a new temp file inside $TMPDIR if set, otherwise inside /tmp.
- TempFile/TempDir::new_in(path) creates a new temp file inside an already existing valid path.
- TempFile/TempDir::new_with_prefix(prefix) is the old TempFile/TempDir::new(prefix) method.

jiangliu commented:
There are some intelj related files included, please help to remove them.

andreeaflorescu commented on src/tempdir.rs:
```diff
@@ -5,13 +5,14 @@
 use std::os::unix::ffi::OsStringExt;
 use std::path::{Path, PathBuf};
 
+use libc;
```
nit: why not have this where it previously was?

iulianbarbu replied:
I structured the `use`s in the following groups: `std` usages in one group, other libs usages in another, same crate modules usages respectively in another.

iulianbarbu commented:
> There are some intelj related files included, please help to remove them.

@jiangliu , removed the intelj files. 

> Please also add more descriptive commit message.

@andreeaflorescu , updated.

Please take a look [here](https://github.com/rust-vmm/vmm-sys-util/commit/3b94a8b57d1d13babb3965a9990e45143ab62129).
